collector/lib/CollectorConfig.h

```diff
@@ -193,12 +200,15 @@ class CollectorConfig {
   std::optional<TlsConfig> tls_config_;
 
   std::optional<sensor::CollectorConfig> runtime_config_;
+  StoppableThread thread_;
+  std::mutex* mutex_ = new std::mutex();
```

Comment:
I don't know where this mutex should live. It is used by `CollectorConfig` and `NetworkStatusNotifier`. It will likely be used by other components in the future.

Reply:
This should probably be a shared pointer.

Reply:
Done